### PR TITLE
use removeObjectIfExists instead removeObject, it is retryable  

### PR DESCRIPTION
**Pull request by CheSema**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
The methods `removeObject` and `removeObjects` are not idempotent. When retries happen due to network errors, the result could be `object not found` because it has been deleted at previous attempts.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll3** [comment]
*This is an automated comment for commit 338af374d88c134b39d75dd1f56f5630cd41fcc2 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/71529/338af374d88c134b39d75dd1f56f5630cd41fcc2/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: src/Disks/ObjectStorages/Local/LocalObjectStorage.h [hunk 1]
```diff
@@ -42,9 +42,9 @@ class LocalObjectStorage : public IObjectStorage
         size_t buf_size = DBMS_DEFAULT_BUFFER_SIZE,
         const WriteSettings & write_settings = {}) override;
 
-    void removeObject(const StoredObject & object) override;
+    void removeObject(const StoredObject & object);
 
-    void removeObjects(const StoredObjects &  objects) override;
```
Comment (divanik) [inline src/Disks/ObjectStorages/Local/LocalObjectStorage.h:47]: Make these functions private

File: src/Disks/ObjectStorages/Local/LocalObjectStorage.h [hunk 2]
```diff
@@ -42,9 +42,9 @@ class LocalObjectStorage : public IObjectStorage
...
+    void removeObjects(const StoredObjects &  objects);
```
Comment (divanik) [inline src/Disks/ObjectStorages/Local/LocalObjectStorage.h:47]: Make these functions private

File: src/Disks/ObjectStorages/HDFS/HDFSObjectStorage.h [hunk 3]
```diff
@@ -78,9 +78,9 @@ class HDFSObjectStorage : public IObjectStorage
         const WriteSettings & write_settings = {}) override;
 
     /// Remove file. Throws exception if file doesn't exists or it's a directory.
-    void removeObject(const StoredObject & object) override;
+    void removeObject(const StoredObject & object);
 
-    void removeObjects(const StoredObjects & objects) override;
+    void removeObjects(const StoredObjects & objects);
```
Comment (divanik) [inline src/Disks/ObjectStorages/HDFS/HDFSObjectStorage.h:83]: Make these functions private

File: src/Disks/ObjectStorages/Local/LocalObjectStorage.h (hunk 2 above)
Comment (divanik) [reply src/Disks/ObjectStorages/Local/LocalObjectStorage.h:47]: Also add const or //NOLIT to definition to suppress clang-tidy

**Review by divanik — COMMENTED** [review]
The PR requires minor fixes

**Review by divanik — APPROVED** [review]

File: src/Disks/ObjectStorages/IObjectStorage.h [hunk 4]
```diff
@@ -161,11 +161,11 @@ class IObjectStorage
     virtual bool isRemote() const = 0;
 
     /// Remove object. Throws exception if object doesn't exists.
-    virtual void removeObject(const StoredObject & object) = 0;
+    // virtual void removeObject(const StoredObject & object) = 0;
 
     /// Remove multiple objects. Some object storages can do batch remove in a more
     /// optimal way.
-    virtual void removeObjects(const StoredObjects & objects) = 0;
+    // virtual void removeObjects(const StoredObjects & objects) = 0;
```
Comment (tavplubix) [inline src/Disks/ObjectStorages/IObjectStorage.h:168]: remove?

**Pull request merged**